homeassistant/components/iammeter/sensor.py

```diff
@@ -0,0 +1,123 @@
+"""Support for iammeter via local API."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from iammeter import real_time_api
+from iammeter.power_meter import IamMeterError
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 80
+DEFAULT_DEVICE_NAME = "IamMeter"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_DEVICE_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+PLATFORM_TIMEOUT = 8
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform setup."""
+    try:
+        with async_timeout.timeout(PLATFORM_TIMEOUT):
+            api = await real_time_api(config[CONF_HOST], config[CONF_PORT])
+    except asyncio.TimeoutError:
+        _LOGGER.error("IamMeter device not ready.")
```

Comment:
Please do not name the integration in log messages, see the [style guidelines](https://developers.home-assistant.io/docs/en/next/development_guidelines.html#log-messages) for more information. Maybe print out the error as well.
```suggestion
        _LOGGER.error("Device is not ready.")
```